version: "3"
volumes:
  dbbackups:
  postgis-data:
  redis:
services:
  main:
    image: patreon
    expose:
      - "8081"
      - "8080"
    ports:
      - "8080:8080"
    networks:
      - default
    depends_on:
      - redis
      - patreon-bd
  redis:
    image: "redis:alpine"
    expose:
      - "6379"
    ports:
      - "6379:6379"
    volumes:
      - redis:/data
  patreon-bd:
    image: postgres:latest
    expose:
      - "5432"
    volumes:
      - ./scripts/user.sql:/docker-entrypoint-initdb.d/init.sql
      - postgis-data:/var/lib/postgresql
    environment:
      - POSTGRES_PASSWORD=pyaterochka
      - POSTGRES_USER=patreon
      - POSTGRES_DB=restapi_dev
    ports:
      - "5432:5432"
    restart: on-failure
    healthcheck:
      test: "exit 0"
    deploy:
      resources:
        limits:
          cpus: '2'
          memory: 4G
#  pgadmin:
#    container_name: pgadmin_container
#    image: dpage/pgadmin4:5.7
#    environment:
#      PGADMIN_DEFAULT_EMAIL: "m@mail.ru"
#      PGADMIN_DEFAULT_PASSWORD: "pyaterochka"
#      PGADMIN_CONFIG_SERVER_MODE: "False"
#    volumes:
#      - ./pgadmin:/var/lib/pgadmin
#    ports:
#      - "5050:80"
#    restart: unless-stopped
#    deploy:
#      resources:
#        limits:
#          cpus: '0.5'
#          memory: 1G
#    networks:
#      - postgres
networks:
  postgres:
    driver: bridge
